### PR TITLE
Adding smoothscroll polyfill for .scroll* methods

### PR DESCRIPTION
**Pull request by nicksrandall**
This feature adds support for "smooth scrolling" when calling methods like `.scrollTo`, `.scroll`, and `.scrollBy` on either the window or a particular DOM element.

Uses https://github.com/iamdustan/smoothscroll

### CODE REVIEW

File: polyfills/smoothscroll/detect.js [hunk 1]
```diff
@@ -0,0 +1 @@
+'scrollBehavior' in d.documentElement.style
```
Comment (JakeChampion) [inline polyfills/smoothscroll/detect.js]: Is d meant to be document?
```suggestion
'scrollBehavior' in document.documentElement.style
```

Comment (nicksrandall) [reply polyfills/smoothscroll/detect.js]: Whoops, yes.

File: polyfills/smoothscroll/config.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+dependencies = [
+  "requestAnimationFrame"
+]
+license = "MIT"
+spec = "https://drafts.csswg.org/cssom-view/#smooth-scrolling"
+repo = "https://github.com/webcomponents/polyfills/tree/master/packages/template"
```
Comment (JakeChampion) [inline polyfills/smoothscroll/config.toml]:
```suggestion
repo = "https://github.com/iamdustan/smoothscroll"
```

**Comment by JakeChampion** [comment]
Thank for the pull-request @nicksrandall, I've added some basic tests for the polyfills and updated the aliases and repo sections in the config file 👍

**Comment by JakeChampion** [comment]
@nicksrandall it looks like the basic tests I've added are failing in some browsers -- https://circleci.com/gh/Financial-Times/polyfill-library/6114
These are the browsers that the tests are failing on:
Chrome 29 to 60
Edge 15 to 18
IE 8 to 11
Safari 9 to 12

**Comment by nicksrandall** [comment]
@JakeChampion I cleaned up the polyfill a bit by getting rid of the UMD code and I re-wrote the tests to use HTMLElement (if available) instead of Element directly. I'm not sure why the tests are failing at this point because CI doesn't give me an error message.

**Comment by fvonellerts** [comment]
Has this been added to polyfill.io?

**Comment by nicksrandall** [comment]
Not yet. @JakeChampion do have any kind of a release cadence for when this will go out?

**Comment by JakeChampion** [comment]
I'm releasing it all now :-) https://github.com/Financial-Times/polyfill-library/releases/tag/v3.42.0

**Pull request merged**